### PR TITLE
Support source connection parameter in JDBC driver

### PR DESCRIPTION
**Pull request by posulliv**
Resolves: #3045

### CODE REVIEW

File: presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoConnection.java [hunk 1]
```diff
@@ -805,4 +797,23 @@ private static String getIsolationLevel(int level)
         }
         throw new SQLException("Invalid transaction isolation level: " + level);
     }
+
+    private String getSource()
```
Comment (kokosing) [inline presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoConnection.java:801]: I find it a bit unintuitive that `getSource` does not simply return a `source` value. Maybe `Source` could get a default value that could evaluated using the below code?

File: presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcConnection.java [hunk 2]
```diff
@@ -289,6 +289,29 @@ public void testApplicationName()
...
+
+        try (Connection connection = createConnection("source=testing")) {
+            connection.setClientInfo("ApplicationName", "testingApplicationName");
+            assertConnectionSource(connection, "testing");
+        }
+
+        try (Connection connection = createConnection("source=testing&applicationNamePrefix=fruit:")) {
+            connection.setClientInfo("ApplicationName", "testingApplicationName");
+            assertConnectionSource(connection, "testing");
+        }
```
Comment (kokosing) [inline presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcConnection.java:312]: can you please also add tests for `assertConnectionSource(connection, "testing");` where `source` parameter is not given in connection string.

File: presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoConnection.java (hunk 1 above)
Comment (posulliv) [reply presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoConnection.java:801]: @kokosing thank you for the feedback!

`ConnectionProperty` does have a `getDefault` method but the issue I see with this is the logic to construct `source` depends on another connection property (`applicationNamePrefix`) and I don't see a way to access other connection properties from within a `ConnectionProperty` instance.

Is this what you were thinking? Is there a way to do this that I am missing?

Comment (kokosing) [reply presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoConnection.java:801]: Ok, then let's rename the method. `getActualSource`?

File: presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcConnection.java (hunk 2 above)
Comment (posulliv) [reply presto-jdbc/src/test/java/io/prestosql/jdbc/TestJdbcConnection.java:312]: yes, done.

File: presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoConnection.java (hunk 1 above)
Comment (posulliv) [reply presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoConnection.java]: Sounds good. Updated.

**Comment by posulliv** [comment]
@kokosing i have applied your feedback. If you have a chance to review again, that would be great!

**Comment by electrum** [comment]
This looks good. Please update documentation: https://github.com/prestosql/presto/blob/master/presto-docs/src/main/sphinx/installation/jdbc.rst

It’s important to mention that this is used in preference to `ApplicationName`.

**Comment by posulliv** [comment]
@electrum sure thing. I updated the docs; let me know if my docs update is not clear.

**Review by kokosing — APPROVED** [review]
@electrum do you want to take a look?

**Review by electrum — APPROVED** [review]
Apologies, I missed following up on this

**Comment by electrum** [comment]
Thanks!

**Pull request merged**